src/javascript/app_2/App/Components/Elements/popconfirm.jsx

```diff
@@ -0,0 +1,88 @@
+import React        from 'react';
+import classNames   from 'classnames';
+import PropTypes    from 'prop-types';
+import { localize } from '../../../../_common/localize';
+
+class PopConfirm extends React.Component {
+    state = {
+        is_open: false,
+    };
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    setWrapperRef = (node) => {
+        this.wrapper_ref = node;
+    }
+
+    handleClickOutside = (event) => {
+        if (this.wrapper_ref && !this.wrapper_ref.contains(event.target) && this.state.is_open) {
+            this.setState({ is_open: false });
+        }
+    }
+
+    handleClose = () => {
+        this.setState({ is_open: false });
+    }
+
+    render() {
+        const popconfirm_class = classNames('popconfirm', this.props.alignment, {
+            'open': this.state.is_open,
+        });
+
+        const popconfirm_element = (
```

Comment:
@zakibinary 
Regarding the Style Guide, it's better to move `popconfirm_element` to another file as a stateless component.

Reply:
Noted, but do you think it's alright if we keep in the same file? As the `popconfirm_element` is only used here.

Reply:
@zakibinary 
For keeping files more readable, it's better to define each component in a separate file.